backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -122,4 +124,25 @@ public UpdateShopStatusResponse updateShop(UpdateShopStatusRequest updateShopSta
 		shopMap.get(updateShopStatusRequest.getShopId()).setStatus(updateShopStatusRequest.getShopStatus());
 		return new UpdateShopStatusResponse(shopMap.get(updateShopStatusRequest.getShopId()));
 	}
+
+	public Shop getShop(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			return shopMap.get(shopId);
+		}
+		// TODO: Throw exception here.
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to fetch information of a shop which is not in the ACTIVE state.");
+		return new Shop();
+	}
+
+	public AtomicInteger getCustomersInQueue(UUID shopId) {
+		List<Token> tokenList = shopIdToListOfTokensMap.get(shopId);
+		AtomicInteger customersInQueue = new AtomicInteger(0);
+		tokenList.stream().forEach(token -> {
+			if (token.getStatus() == Status.ACTIVE) {
+				customersInQueue.getAndIncrement();
+			}
+		});
```

Comment:
We can maybe use something like: 
tokenList.stream()
                            .filter(x -> condition checking status active))
                            .count();

Reply:
Ok changed it